New notebook to showcase how SageMaker Clarify works with BYOC

xgchena opened the pull request:
*Issue #, if available:*

N/A

*Description of changes:*

As indicated from customer feedback, a customer pain-point of using SageMaker Clarify is how to properly connect SageMaker Clarify jobs with their own containers. Without fully understanding the contracts between them, it is time-consuming for customers to run Clarify job successfully by trial-and-error. This commit adds a new notebook to demonstrate how to build a custom container that can work seamlessly with SageMaker Clarify jobs. Hope this can settle customers' doubts.

**Testing Done**

* Tested the new notebook by manual full-run in us-west-2 and us-east-1.
* Built doc locally and checked that the new notebook shows up in the "Processing" page.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

xgchena commented:
The notebook uses the [SageMaker Docker Build CLI tool](https://github.com/aws-samples/sagemaker-studio-image-build-cli) to build the custom container. The check "AWS CodeBuild us-west-2 (amazon-sagemaker-examples-pr)" failed because the role it used to execute the notebook doesn't have necessary permissions to run the tool, thus the customer container was not built, and then training custom model failed due to missing container image.

```
---------------------------------------------------------------------------
Exception encountered at "In [20]":
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
<ipython-input-20-e70b6adae38a> in <module>
      6     sagemaker_session=session,
      7 )
----> 8 estimator.fit({'train': train_input}, logs=False)

/opt/conda/lib/python3.7/site-packages/sagemaker/estimator.py in fit(self, inputs, wait, logs, job_name, experiment_config)
    659         self._prepare_for_training(job_name=job_name)
    660 
--> 661   

[...]

ClientError: An error occurred (ValidationException) when calling the CreateTrainingJob operation: Cannot find repository: sagemaker-studio-byoc in registry ID: 521695447989 Please check if your ECR repository exists and role arn:aws:iam::521695447989:role/SageMakerRole has proper pull permissions for SageMaker: ecr:BatchCheckLayerAvailability, ecr:BatchGetImage, ecr:GetDownloadUrlForLayer
```

mchoi8739 commented on sagemaker_processing/fairness_and_explainability/fairness_and_explainability_byoc.ipynb:
```diff
@@ -0,0 +1,1204 @@
+{
```
<p>You can include the following cell to make it easy to find:</p><pre class="ql-syntax" spellcheck="false">! aws s3 cp {bias_report_output_path} ./ --recursive
</pre><p>and then open the file:</p><pre class="ql-syntax" spellcheck="false">import IPython
IPython.display.HTML(filename='bias_report.html')
</pre><p> </p><p>Change the file name with the correct one. I just made it up as an example.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2111/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='77'/>

xgchena replied:
Thanks for the tip, I added the code snippet to the notebook and the html reports are displayed inline perfectly.

xgchena commented:
Updates:
* Addressed review comments
* Fixed two code formatting issues
* Corrected several grammar problems reported by the "sagemaker-examples-grammar" check. (The remaining "problems" are false alarms)
* Re-tested the new notebook by manual full-run in us-west-2
